packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx

```diff
@@ -0,0 +1,230 @@
+  });
+
+  if (!currentUser) return <></>;
+  const preTopicRecommendations: {comment: {_id: string, contents: {html: string, plaintextMainText: string}}, recommendationReason: string, yourVote: string, theirVote: string}[] | undefined = topicData?.GetTwoUserTopicRecommendations; 
+  const topicRecommendations = preTopicRecommendations?.filter(topic => topic.theirVote !== null);
+  const numRecommendations = topicRecommendations?.length || 0;
+  const numShown = isExpanded ? numRecommendations : 1
+  const numHidden = Math.max(0, numRecommendations - numShown);
+
+  return (
+    <div>
+      <div key={targetUser._id} className={ isExpanded ? classNames(classes.dialogueUserRow, classes.dialogueUserRowExpandedMobile) : classes.dialogueUserRow}>
+        <div className={ (isExpanded || numRecommendations === 0) ? classNames(classes.dialogueLeftContainer, classes.dialogueLeftContainerExpandedMobile, classes.dialogueLeftContainerNoTopics) : classes.dialogueLeftContainer}>
+          <div className={ classes.dialogueMatchCheckbox }>
+            <DialogueCheckBox
+              targetUserId={targetUser._id}
+              targetUserDisplayName={targetUser.displayName}
+              checkId={checkId}
+              isChecked={userIsChecked}
+              isMatched={userIsMatched}
+            />
+          </div>
+          <PostsItem2MetaInfo className={classes.dialogueMatchUsername}>
+            <UsersName
+              documentId={targetUser._id}
+              simple={false}
+            />
+          </PostsItem2MetaInfo>
+        </div>
+        {showSuggestedTopics && <div className={classes.topicRecommendationsList}>
+          {topicRecommendations?.slice(0,numShown).map((topic, index) => (
+            <div key={index} className={classes.suggestionRow}>
+              <p key={index} className={ isExpanded ? classes.debateTopicExpanded : classes.debateTopicCollapsed}>
+                {topic.theirVote === 'agree' && [<ReactionIcon key={index} size={13} react={"agree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>agrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] } 
+                {topic.theirVote === 'disagree' && [<ReactionIcon key={index} size={13} react={"disagree"} />, <span key={index} className={ isExpanded ? classes.agreeText : classNames(classes.agreeText, classes.agreeTextCollapsedMobile) }>disagrees: </span>, `"${topic.comment.contents.plaintextMainText}"`] }
+              </p>
+            </div>
+          ))}
+          
+      </div>}
+      <div className={classes.dialogueRightContainer}>
+        {<span className={isExpanded ? classes.hideIcon : classes.expandIcon} onClick={toggleExpansion}>
+          {isExpanded ? '▲ hide' : (numHidden > 0 ? [`▼`, <span key={targetUser._id} className={classes.bigScreenExpandNote}> ...more</span>] : '')} 
```

Comment:
Nested ternaries are very difficult to read.  Please extract into an assignment to a variable before the `return` (which can just call a function with 2 if statements, or something)